In [1]:
import numpy as np

def rule3 (da,db):
    dQ = np.sqrt(da**2+db**2)
    return dQ

d1 = 1.0
d2 = 0.3
errd1 = 0.0005
errd2 = 0.0005
dtotal= d1+d2
errdtotal = rule3(errd1, errd2)

print("dtotal=",dtotal)
print("errdtotal=",errdtotal)

dtotal= 1.3
errdtotal= 0.0007071067811865475


In [6]:
def rule1 (da,c):
    dQ = (c*da)
    return dQ

da = .005
c = 3
dQ = rule1(c,da)
print (dQ)



0.015


In [3]:
x = np.array([1.1,1.3,1.4,.9,.95,1.05])
xmean = np.average(x)
xstdev = np.std(x)
print(xmean)
print(xstdev)

1.1166666666666667
0.17950549357115014


The average of 1.1, 1.3, 1.4, .9, .95, 1.05 is 1.1167 and the standard deviation is 0.1795

$\delta Q = \sqrt{\delta A^2 + \delta B^2} $

$\delta v = v \sqrt{(\frac{1\delta x}{x})^2 + (\frac{-.5
\delta y}{y})^2} $

In [8]:
def rule2 (c,m,A,da):
    dQ = (c*m*A**(m-1)*da)
    return dQ

c = 2
m = 5
A = 4
da = .005
dQ = rule2(c,m,A,da)
print (dQ)

12.8


In [18]:
def rule4 (Q,m,n,da,db,A,B):
    dQ = Q*(np.sqrt(((m*da)/A)**2 + ((n*db)/B)**2))
    return dQ
Q=2
m = 1
n = 2
da = .05
db=.01
A = 5
B = 7
dQ = rule4 (Q,m,n,da,db,A,B)
print (dQ)

0.02080031396937291


In [20]:
def rule4(v,a,dM,b,dm,m,c,dR,R,d,dh,h):
    dv= v*np.sqrt(((a*dM)/M)**2 + ((b*dm)/m)**2 + ((c*dR)/R)**2 + ((d*dh)/h)**2)
    return dv
v = 2.384
a = 1
dM = .0001
M = .2578
b = -1 
dm = .0001
m = 0.0659
c = 1/2
dR = 0.005
R = .28
d = .5
dh = .000516
h = .06768

dv = rule4 (v,a,dM,b,dm,m,c,dR,R,d,dh,h)
print (dv)


0.023443859565717474
